public/app/algorithms/two_anonimize_geolocations.py

```diff
@@ -0,0 +1,89 @@
+''' Mask country, region and city attributes when necessary so that no user can be uniquely identified '''
+
+import json, argparse
+
+def load_dataset(dataset_path):
+    ''' Load a json dataset from the specified path '''
+    with open(dataset_path) as json_file:
+        return json.load(json_file)
+
+def save_dataset(dataset, dest_path):
+    ''' Save a json dataset to the specified path '''
+    with open(dest_path, 'w') as dest_file:
+        json.dump(dataset, dest_file, indent=2)
+
+def parse_arguments():
+    ''' Parses command-line arguments. '''
+    parser = argparse.ArgumentParser(description='Mask country, region and city attributes when necessary so that no user can be uniquely identified')
+    parser.add_argument('dataset_path', help='path to the json dataset')
+    parser.add_argument('output_path', default='output.json',
+        help='destination path (including file name) for the output')
+
+    return parser.parse_args()
+
+def main():
+    args = parse_arguments()
+
+    dataset = load_dataset(args.dataset_path)
+
+    actions = dataset['data']['actions']
+
+    # get first geolocation for each user
+    geolocations = {}
+    for action in actions:
+        user = action['user']
+        if user not in geolocations:
+            geolocations[user] = action['geolocation']
+            geolocations[user]['range'] = [0,0]
```

Comment:
so this file is a combination between k-anonymization and suppression (we are effectively deleting all these fields that we are emptying). maybe add to the algorithm description: _, with remaining geolocation information suppressed_